### PR TITLE
Minghe/dhcpcd and cloud init

### PR DESCRIPTION
**Pull request by rmhsawyer**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add [dhcpcd](https://github.com/NetworkConfiguration/dhcpcd) package to replace dhcp package as it is deprecated upstream. Correspondingly, upgrade current version of cloud-init from 23.4 to [24.2](https://github.com/canonical/cloud-init/releases/tag/24.2) to add dhcpcd, azurelinux distro support.

###### Change Log  

- Add dhcpcd package. Import spec file from [Fedora dhcpcd.spec](https://src.fedoraproject.org/rpms/dhcpcd/blob/rawhide/f/dhcpcd.spec)
- Upgrade cloud-init to 24.2. Remove unnecessary patch files as they are included in the new version of cloud-init

###### Does this affect the toolchain?  

NO

###### Associated issues  

- [Bug 51901579](https://microsoft.visualstudio.com/OS/_workitems/edit/51901579): [Core] [dhcp] Evaluate dhcpcd as an alternative to ISC DHCP, which is deprecated upstream since 2022

###### Test Methodology

- Pipeline build id: xxxx
- Buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=599111&view=results
- Local package and image build
- Azure provisioning test. Verify VM successfully deployed and cloud-init status is running OK

### CODE REVIEW

File: SPECS/dhcpcd/dhcpcd.spec [hunk 1]
```diff
@@ -0,0 +1,83 @@
+%global debug_package   %{nil}
+
+Summary:        A minimalistic network configuration daemon with DHCPv4, rdisc and DHCPv6 support
+Name:           dhcpcd
+Version:        10.0.8
+Release:        1%{?dist}
+License:        BSD-2-Clause AND ISC AND MIT
+Url:            http://roy.marples.name/projects/%{name}/
+Source0:        https://github.com/NetworkConfiguration/%{name}/archive/refs/tags/v%{version}.tar.gz
```
Comment (christopherco) [inline SPECS/dhcpcd/dhcpcd.spec:9]:
```suggestion
Source0:        https://github.com/NetworkConfiguration/%{name}/archive/refs/tags/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
```

**Review by christopherco — APPROVED** [review]

File: SPECS/dhcpcd/dhcpcd.spec [hunk 2]
```diff
@@ -0,0 +1,83 @@
+%global debug_package   %{nil}
```
Comment (christopherco) [inline SPECS/dhcpcd/dhcpcd.spec:1]: Do we really need to disable the debuginfo build for this package? Was there an error that caused the need for this line?

File: SPECS/dhcpcd/dhcpcd.spec [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+%{_unitdir}/%{name}.service
+%{_unitdir}/%{name}@.service
+%defattr(0644,root,dhcpcd,0755)
+%{_sharedstatedir}/%{name}
+
+%changelog
+* Tue Jun 25 2024 Minghe Ren <mingheren@microsoft.com> 10.0.8-1
+- Initial build
+- Initial Azure Linux import from Fedora 41 (license: MIT)
+- License verified
```
Comment (christopherco) [inline SPECS/dhcpcd/dhcpcd.spec:83]:
```suggestion
- License verified

```
nit - should have newline at end of file

**Review by tobiasb-ms — APPROVED** [review]

File: SPECS/dhcpcd/dhcpcd.spec (hunk 3 above)
Comment (rmhsawyer) [reply SPECS/dhcpcd/dhcpcd.spec:83]: change as suggested

File: SPECS/dhcpcd/dhcpcd.spec (hunk 2 above)
Comment (rmhsawyer) [reply SPECS/dhcpcd/dhcpcd.spec]: Initial added this for a build error. After rebuilding looks like we don't need it and removed

**Pull request merged**